Handlers/ProxyUserIpAuditTrailEventHandler.cs

```diff
@@ -0,0 +1,29 @@
+using Microsoft.AspNetCore.Http;
+using OrchardCore.AuditTrail.Services;
+using OrchardCore.AuditTrail.Services.Models;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Lombiq.AuditTrailExtensions.Handlers
+{
+    public class ProxyUserIpAuditTrailEventHandler : AuditTrailEventHandlerBase
+    {
+        private readonly IHttpContextAccessor _hca;
+
+        public ProxyUserIpAuditTrailEventHandler(IHttpContextAccessor hca) => _hca = hca;
+
+        public override Task CreateAsync(AuditTrailCreateContext context)
+        {
+            if (_hca?.HttpContext?.Request?.Headers.ContainsKey("X-Forwarded-For") == true)
```

Comment:
Nem lesz ez így is NRE, mivel a `ContainsKey()` előtt nincs `?`?

Reply:
![kép](https://user-images.githubusercontent.com/19938236/126461271-71811b71-8226-47f5-93c6-034be84eddc8.png)

Reply:
Ha a Request nem null akkor garantáltan a Request.Header se mert létrehozáskor rögtön kap értéket:
![image](https://user-images.githubusercontent.com/4007293/126462849-845a4bfe-e618-456c-93ac-a244a131081e.png)

Reply:
Eskü ezért nem raktam kérdőjelet. :)

Reply:
Ah cool, köszi Krisz, hogy belenéztél a .NET Core source-ába! :D